<!-- -->
<launch>

  <param name="use_sim_time" value="true"/>

<!--
  <node pkg="urg_node" type="urg_node" name="front_hokuyo" output="screen" respawn="true">
    <param name="serial_port"    value="/dev/ttyACM1"/>
    <param name="serial_baud"    value="115200"/>

    <param name="frame_id"       value="front_scanner"/>

    <param name="calibrate_time"     value="true"/>
    <param name="publish_intensity"  value="false"/>
    <param name="publish_multiecho"  value="false"/>

    <param name="min_ang"        value="-2.0862"/>
    <param name="max_ang"        value="2.0923"/>

    <param name="skip"           value="0"/>
    <param name="time_offset"    value="0.0"/>
    <param name="cluster"        value="1"/>

    <remap from="scan" to="front_scan"/>
  </node>

  <node pkg="urg_node" type="urg_node" name="rear_hokuyo" output="screen" respawn="true">
    <param name="serial_port"    value="/dev/ttyACM0"/>
    <param name="serial_baud"    value="115200"/>

    <param name="frame_id"       value="rear_scanner"/>

    <param name="calibrate_time"     value="true"/>
    <param name="publish_intensity"  value="false"/>
    <param name="publish_multiecho"  value="false"/>

    <param name="min_ang"        value="-2.0862"/>
    <param name="max_ang"        value="2.0923"/>

    <param name="skip"           value="0"/>
    <param name="time_offset"    value="0.0"/>
    <param name="cluster"        value="1"/>

    <remap from="scan" to="rear_scan"/>
  </node>
-->


  <!--node pkg="obstacle_detector" type="scans_merger_node" name="scans_merger">
    <param name="active"            value="true"/>
    <param name="publish_scan"      value="true"/>
    <param name="publish_pcl"       value="false"/>

    <param name="ranges_num"        value="1000"/>

    <param name="min_scanner_range" value="0.05"/>
    <param name="max_scanner_range" value="10.0"/>

    <param name="min_x_range"       value="-10.0"/>
    <param name="max_x_range"       value="10.0"/>
    <param name="min_y_range"       value="-10.0"/>
    <param name="max_y_range"       value="10.0"/>

    <param name="frame_id"          value="scanner_base"/>
  </node-->

  <node pkg="obstacle_detector" type="obstacle_extractor_node" name="obstacle_extractor" output="screen" respawn="true">
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>
    <param name="use_split_and_merge"  value="true"/>
    <param name="discard_converted_segments" value="true"/>
    <param name="transform_coordinates"      value="true"/>

    <param name="min_group_points"     value="5"/>

    <param name="max_group_distance"   value="0.100"/>
    <param name="distance_proportion"  value="0.006136"/>
    <param name="max_split_distance"   value="0.070"/>
    <param name="max_merge_separation" value="0.150"/>
    <param name="max_merge_spread"     value="0.070"/>
    <param name="max_circle_radius"    value="0.500"/>
    <param name="radius_enlargement"   value="0.10"/>

    <param name="frame_id"             value="base_footprint"/>
  </node>

  <node pkg="obstacle_detector" type="obstacle_tracker_node" name="obstacle_tracker">
    <param name="active"                  value="true"/>

    <param name="loop_rate"               value="100.0"/>
    <param name="tracking_duration"       value="2.0"/>
    <param name="min_correspondence_cost" value="0.6"/>
    <param name="std_correspondence_dev"  value="0.15"/>
    <param name="process_variance"        value="0.1"/>
    <param name="process_rate_variance"   value="0.1"/>
    <param name="measurement_variance"    value="1.0"/>

    <remap from="tracked_obstacles" to="obstacles"/>
  </node>

  <node pkg="obstacle_detector" type="obstacle_publisher_node" name="obstacle_publisher">
    <param name="active"           value="false"/>
    <param name="reset"            value="false"/>
    <param name="fusion_example"   value="false"/>
    <param name="fission_example"  value="false"/>

    <param name="loop_rate"        value="100.0"/>

    <rosparam param="x_vector">[-1.0, 0.0, 1.0]</rosparam>
    <rosparam param="y_vector">[0.0, -1.0, 0.0]</rosparam>
    <rosparam param="r_vector">[0.5, 0.5, 0.5]</rosparam>
    <rosparam param="vx_vector">[0.0, 0.0, 0.0]</rosparam>
    <rosparam param="vy_vector">[0.0, 0.0, 0.0]</rosparam>

    <param name="frame_id"         value="map"/>
    <remap from="obstacles" to="raw_obstacles"/>
  </node>

  <node pkg="obstacle_detector" type="obstacle_recorder_node" name="obstacle_recorder">
    <param name="active"                  value="true"/>
    <param name="recording"               value="false"/>

    <param name="filename_prefix"         value="tracked_"/>
  </node>

  <node pkg="obstacle_detector" type="obstacle_map" name="obstacle_map" output="screen"/>

  <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find fira6_navigation)/rviz/test.rviz"/>

</launch>
